Refactor

SaxanLosterToreda opened the pull request:
I'm trying to reduce the complexity and improve the intuitive understanding of the names of method.

SaxanLosterToreda committed:
Some basic functionality and map out tests

SaxanLosterToreda committed:
testing for logger started

SaxanLosterToreda committed:
finished logger tests

SaxanLosterToreda committed:
log-listener tests

SaxanLosterToreda committed:
use handleMessage() to put message in logs, user action() does something with logs on showLogs()

SaxanLosterToreda committed:
cleaned up init and props using options/state interfaces for both Logger and LogListener

SaxanLosterToreda committed:
random ids for logger and logevents

SaxanLosterToreda committed:
Switched from using emits and listeners to using transports

SaxanLosterToreda committed:
Functional, but having trouble with testing async execute function

SaxanLosterToreda committed:
dumb errors that I didn't catch before push

SaxanLosterToreda committed:
finished async test

SaxanLosterToreda committed:
a few fixes to tests

SaxanLosterToreda committed:
sonar scanner update

SaxanLosterToreda committed:
some fixes to condition coverage

SaxanLosterToreda committed:
created a default action for transports

SaxanLosterToreda committed:
Remove old files

SaxanLosterToreda committed:
sonarcloud suggested changes

SaxanLosterToreda committed:
change bitwise lint from error to warning in this project.

SaxanLosterToreda committed:
more linting errors

SaxanLosterToreda committed:
more linting

SaxanLosterToreda committed:
Updated the README to show usage and added some functions to make Logger easier to use.

SaxanLosterToreda committed:
Final? change

SaxanLosterToreda committed:
Merge pull request #2 from toreda/feature/file-logging

Feature/file logging

mbrich committed:
Removed tslint in favor of eslint using toreda org rules for typescript.
Updated package naming.
Removed prettier.config.js and replaced with toreda package.json key '@toreda/prettier-config'.
Updated license

mbrich committed:
Updated .gitattributes to standardize file line endings.

mbrich committed:
Added example file.
Updated readme examples.

mbrich committed:
Updated github actions to run on develop branch instead of just master.

mbrich committed:
Fixed sonar token secret id in github workflow.

SaxanLosterToreda committed:
didn't pull before fixes so they had to redone

mbrich committed:
Merge pull request #5 from toreda/feature/type-box-fix

merged w/ issues to resolve in develop

mbrich committed:
Added eslint as project dependency with toreda rules.
Changed EOL to LF.
Updated classnames using "Logger" (old package name) to "Log" prefix for consistency. No class changes made beyond rename.

SaxanLosterToreda committed:
Use updated ActionResult and fixed errors from updated tslint

mbrich committed:
Added relative path prefix to sonar.sources and sonar.tests

mbrich committed:
Added public flag to package.json
Moved prettier package to devDependencies

mbrich committed:
Fixed test errors caused by Typescript version upgrade and empty promise resolve calls.

mbrich committed:
Refactoring to add log groups.

mbrich committed:
Cleanup, refactoring, and more tests for log groups.

mbrich committed:
Tests & refactoring for LogTransport
Tests & refactoring for LogState
Removed LogTransportState, which was no longer needed after refactoring.

mbrich committed:
Fixed broken Log and LogTransport tests due to refactoring.

mbrich committed:
Renamed LogTransportAction to LogAction to better convey role.
Simple refactoring to support tests and fix errors found during testing. Expanding tests for groups, transports, and log.

mbrich committed:
Cleanup and fixed more tests across all classes

mbrich committed:
Cleanup and fixed more tests across all classes
Removed LogTransportOptions which is no longer used.
Added init time global and group log level settings with tests.

mbrich committed:
Updated readme
Updated LogLevels to be the expected bitmasks. ALL did not include all previous bits in mask.
Added remove by id methods for transports.

mbrich committed:
Removed `@toreda/action-result` package as its not longer used in project.

mbrich committed:
Updated docs
Added additional helper methods and tests in Log and Group for handling log levels and multiple transport removals.

mbrich committed:
Added additional start options.
Added console logging support which is disabled on start unless specifically enabled by start options.

mbrich committed:
Updated autogenerated docs

mbrich committed:
Updated readme

mbrich committed:
Updated version number for pending release.
Added missing LogActionConsole to package exports.

SaxanLosterToreda committed:
Edit types for better code hinting.
Little bit of formatting of imports

SaxanLosterToreda committed:
Check for options before calling parse.
Update eslint-config.
Use alias paths in test files

SaxanLosterToreda committed:
Update strong-types

# Conflicts:
#	.eslintrc.js
#	src/log.ts
#	src/log/group.ts
#	src/log/state.ts
#	tests/log.spec.ts
#	tests/log/group.spec.ts

SaxanLosterToreda committed:
Merge branch 'master' into develop

SaxanLosterToreda committed:
Merge branch 'master' into develop

SaxanLosterToreda committed:
Removed null check in state.
update yarn.lock after yarn upgrade

SaxanLosterToreda committed:
fixed some convoluted import paths

SaxanLosterToreda committed:
Update package version from 0.4.0 to 0.4.1

SaxanLosterToreda committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	yarn.lock

SaxanLosterToreda committed:
Update `strong-types` to 0.11.1.

SaxanLosterToreda committed:
Minor refactor - mostly fixing naming confusion
 * Changed default group 'global' to 'default'. It makes it more clear how 'all' and 'global' are different.
 * Collapsed multiple convenience functions into singular functions with optional arguments.
 * Renamed convenience logging function (e.g. `errorGroup` to `errorTo`)
 * Reduced size of bitmask in LogLevels. It was bigger than the max size of a js number.
 * Reworked the createMessage function to better handle converting objects to strings.
 * Replaced getGroup with manual test in some functions to prevent auto creation of groups in those functions.

SaxanLosterToreda committed:
Merge branch 'master' into develop

# Conflicts:
#	package.json

SaxanLosterToreda committed:
Added option to create log with specific groups.
Added methods to disable specific log levels.
Added accessor methods to log for enabling/disabling group levels

SaxanLosterToreda committed:
Removed automatic group creation from getGroup.
Overloaded Log.getGroup
 - Calling with no args return the default group.
 - Calling with just a string id return that group if exists or null
 - Calling with nullable id and false return that group if it exists or null
 - Calling with nullable id and true return that group if it exists or the default group

mbrich commented on src/log/levels/helpers.ts:
```diff
@@ -0,0 +1,57 @@
+	obj(obj() | level);
+}
+
+export function LogLevelEnableMultiple(obj: StrongInt, levels: number[]): void {
```
Minor `TypeScript` style issue: for consistency, function names start with a lowercase letter.

- [x] Change `LogLevelEnableMultiple` to `logLevelEnableMultiple`

mbrich commented on src/log/levels/helpers.ts:
```diff
@@ -0,0 +1,57 @@
+import {StrongInt} from '@toreda/strong-types';
+
+export function LogLevelEnable(obj: StrongInt, level: number): void {
```
Minor `TypeScript` style issue: for consistency, function names start with a lowercase letter.

- [x] Change `LogLevelEnable` to `logLevelEnable`

mbrich commented on src/log/levels/helpers.ts:
```diff
@@ -0,0 +1,57 @@
+	}
+}
+
+export function LogLevelDisable(obj: StrongInt, level: number): void {
```
Minor `TypeScript` style issue: for consistency, function names start with a lowercase letter.

- [x]  Change LogLevelDisable to logLevelDisable

mbrich commented on src/log/levels/helpers.ts:
```diff
@@ -0,0 +1,57 @@
+	obj(obj() ^ level);
+}
+
+export function LogLevelDisableMultiple(obj: StrongInt, levels: number[]): void {
```
Minor `TypeScript` style issue: for consistency, function names start with a lowercase letter.

- [x]  Change LogLevelDisableMultiple to logLevelDisableMultiple

SaxanLosterToreda committed:
Log now has a single group attached to it. Making a new group creates a new instance of Log that shares a reference to a global state with the log that created it. The new Log instance has an id that is extended from the Log that created it. Logs can use the global state to change settings in all Logs connected to it or to grab a specific Log for use.

SaxanLosterToreda committed:
Fix import paths

SaxanLosterToreda committed:
changes some options props in LogOptions interface to required

SaxanLosterToreda committed:
ensure all required are present

mbrich reviewed (approved):
Update `README.md` and `CHANGELOG.md` and we're good to go for merge.

SaxanLosterToreda committed:
Cleanup before merge

StrongNumber -> StrongInt.
LogOptions flipped order of union produces better hinting.
Added check in Log for LogStateGlobal and error for improper usage.
Added CHANGELOG and updated README.
Regenerated docs.
